packages/runtime/runtime-utils/src/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -114,6 +170,88 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
         return gcData;
     }
 
+    /**
+     * Called during the start of a summary. Update the work-in-progress used state.
+     */
+    public startSummary(referenceSequenceNumber: number, summaryLogger: ITelemetryLogger) {
+        assert(this.wipUsedState === undefined, "wipUsedState should not be set yet in startSummary");
+        this.wipUsedState = this.used;
+        super.startSummary(referenceSequenceNumber, summaryLogger);
+    }
+
+    /**
+     * Called after summary has been uploaded to the server. Add the work-in-progress state to the pending
+     * summary queue. We track this until we get an ack from the server for this summary.
+     */
+    protected completeSummaryCore(
+        proposalHandle: string,
+        parentPath: EscapedPath | undefined,
+        parentSkipRecursion: boolean,
+    ) {
+        assert(this.wipUsedState !== undefined, "wipUsedState should have been set in startSummary");
+        const wipUsedState = this.wipUsedState;
```

Comment:
nit: suggest to reverse the order here, and then assert against the local variable instead of the property.

Reply:
Done.